File: CSC454HW4/CSC454HW4/src/csc454hw4/EventModelInterface.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package csc454hw4;

/**
 *
 * @author chrisrk192
 */
public interface EventModelInterface {
    String[] OutPut();    
    void StateTransitionExt(String[] inputItems) throws CantPerformAction;
    void StateTransitionInt() throws CantPerformAction;
    void StateTransitionCon(String[] inputItems) throws CantPerformAction;
    double TimeAdvance();
    public String GetStatus();
    
}

File: CSC454HW5/CppImpOfHW4(oldNoWork)/EventModelInterface.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   EventModelInterface.cpp
 * Author: chrisrk192
 * 
 * Created on October 30, 2017, 8:50 PM
 */

#include "EventModelInterface.h"
template<class I, class O>
EventModelInterface<I,O>::EventModelInterface() {
}
template<class I, class O>
EventModelInterface<I,O>::EventModelInterface(const EventModelInterface& orig) {
}
template<class I, class O>
EventModelInterface<I,O>::~EventModelInterface() {
}

template<class I, class O>
O EventModelInterface<I,O>::OutPut(){

}   

template<class I, class O>
void EventModelInterface<I, O>::StateTransitionExt(I inputItems, double e){

}

template<class I, class O>
void EventModelInterface<I, O>::StateTransitionInt(){

}

template<class I, class O>
void EventModelInterface<I, O>::StateTransitionCon(I inputItems){

}

double TimeAdvance(){
}

template<class I, class O>
O EventModelInterface<I, O>::GetStatus(){

}